### ISSUE TITLE
RuntimeError: indirect call signature mismatch in scipy get_blas_funcs on float32 numpy arrays

### ISSUE THREAD

**Issue by ogrisel** [pyodide/pyodide#2763]
## 🐛 Bug

Calling some BLAS operations via scipy's `get_blas_funcs` causes:

```
RuntimeError: indirect call signature mismatch
```

and a crash of the jupyterlite environment (I have to restart the kernel).

The original problem was reported as:

- https://github.com/scikit-learn/scikit-learn/issues/23707

### To Reproduce

Using scipy 1.8.0 from via https://jupyter.org/try-jupyter/lab/ with the following snippet

```python
import numpy as np
from scipy.linalg.blas import get_blas_funcs

arr = np.array([1, 2, 3], dtype=np.float32)
get_blas_funcs('nrm2', dtype=arr.dtype, ilp64=False)(arr)
```

### Expected behavior

This would compute the norm of the array without having to convert it to `np.float64` first.

### Note

When setting `ilp64=True` explicitly, one gets the following Python exception as expected instead of a pyodide crash:

<degails>

```python-traceback
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Input In [4], in <cell line: 5>()
      2 from scipy.linalg.blas import get_blas_funcs
      4 arr = np.array([1, 2, 3], dtype=np.float32)
----> 5 get_blas_funcs('nrm2', dtype=arr.dtype, ilp64=True)(arr)

File /lib/python3.10/site-packages/scipy/linalg/blas.py:400, in _memoize_get_funcs.<locals>.getter(names, arrays, dtype, ilp64)
    397 if value is not None:
    398     return value
--> 400 value = func(names, arrays, dtype, ilp64)
    402 if key is not None:
    403     memo[key] = value

File /lib/python3.10/site-packages/scipy/linalg/blas.py:478, in get_blas_funcs(names, arrays, dtype, ilp64)
    476 else:
    477     if not HAS_ILP64:
--> 478         raise RuntimeError("BLAS ILP64 routine requested, but Scipy "
    479                            "compiled only with 32-bit BLAS")
    480     return _get_funcs(names, arrays, dtype,
    481                       "BLAS", _fblas_64, None, "fblas_64", None,
    482                       _blas_alias, ilp64=True)

RuntimeError: BLAS ILP64 routine requested, but Scipy compiled only with 32-bit BLAS
```

</details>

### Environment

- Pyodide Version: 0.20.0
- Browser version: Firefox 101.0.1

**Comment by rth** [comment]
Thanks for the detailed report! It looks like it was fixed on main. You can try the dev deployment in https://pyodide.org/en/latest/console.html where both this example and the TSNE example works fine. We are planning to do an alpha release this week.

I'm not exactly sure what caused it. As far as I can tell there were no changes in [scipy patches](https://github.com/pyodide/pyodide/commits/main/packages/scipy) that could explain this, though there were updates in numpy and emscripten versions recently.

**Comment by ogrisel** [comment]
Thanks Roman, I confirm it also works on my browser where I reproduced the original problem. Closing.

**Comment by hoodmane** [comment]
This was fixed by #2728.
